legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/h2Extension.pure

```diff
@@ -81,7 +81,7 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::h2::g
     dynaFnToSql('monthNumber',            $allStates,            ^ToSql(format='month(%s)')),
     dynaFnToSql('mostRecentDayOfWeek',    $allStates,            ^ToSql(format='dateadd(DAY, case when %s - DAY_OF_WEEK(%s) > 0 then %s - DAY_OF_WEEK(%s) - 7 else %s - DAY_OF_WEEK(%s) end, %s)', transform={p:String[1..2] | $p->formatMostRecentH2('current_date()')}, parametersWithinWhenClause = [false, false])),
     dynaFnToSql('now',                    $allStates,            ^ToSql(format='current_timestamp()')),
-    dynaFnToSql('parseDate',              $allStates,            ^ToSql(format='parsedatetime(%s,%s)')),
+    dynaFnToSql('parseDate',              $allStates,            ^ToSql(format='parsedatetime(%s,%s)', transform={p:String[*]|$p->transformParseDateParams()})),
```

Comment:
Should this be String[2] ?

Reply:
ideally it should be String[1..2] as we will have a date parameter as string, and a date format which is optional